Mathlib/Combinatorics/Colex.lean

```diff
@@ -305,6 +340,68 @@ lemma lt_iff_max'_mem {s t : Colex α} :
     s < t ↔ ∃ h : s ≠ t, (ofColex s ∆ ofColex t).max' (max_mem_aux h) ∈ ofColex t := by
   rw [lt_iff_le_and_ne, le_iff_max'_mem]; aesop
 
+lemma lt_iff_exists_forall_lt_mem_iff_mem :
+    toColex s < toColex t ↔ ∃ w ∈ t, w ∉ s ∧ ∀ ⦃a⦄, w < a → (a ∈ s ↔ a ∈ t) := by
```

Comment:
Do you think the `∀ ⦃a⦄, w < a → (a ∈ s ↔ a ∈ t)` here would be better as `s.filter (w < .) = t.filter (w < .)`? Or even using
```suggestion
    toColex s < toColex t ↔ ∃ w ∈ t \ s, s.filter (w < .) = t.filter (w < .) := by
```

Reply:
I've tried your suggestion in https://github.com/YaelDillies/LeanCamCombi/blob/master/LeanCamCombi/Mathlib/Combinatorics/Colex.lean but somehow it makes everything more painful. It seems reasoning about colex really is done element-wise.

Reply:
Did you mean to link to a different branch that contained this suggestion?

Reply:
No sorry, I only changed things in local because it was pretty clear pretty quickly that your approach wouldn't work

Reply:
Would a `s.filter P = t.filter P ↔ ∀ ⦃a⦄, P a → (a ∈ s ↔ a ∈ t)` lemma make my suggestion more pleasant? I just think it's easier to read, even if it doesn't help with proofs.

Reply:
Sure, I guess I can do that

Reply:
What's the status of this conversation?

Reply:
Sorry, forgot to resolve